torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -466,6 +466,18 @@ RegisterOperators reg({
             };
           }
         }),
+    // TODO schema
+    Operator(prim::Fork, [](Node* stack) -> Operation {
+      return [=](Stack& stack) -> int {
+        throw detail::NewCoroutine();
+      };
+    }),
+    // TODO: type variable
+    Operator(prim::Wait, [](Node* stack) -> Operation {
```

Comment:
We should be able to simply register wait as a builtin with schema:
```
"aten::wait(Future[t] self) -> t"
```
This way we do not need tons of ugly SugaredValue handling for it.